tests/test_core.py

```diff
@@ -1502,6 +1797,31 @@ def _event_listener(event):
 
     hass.states.async_set("light.bedroom", "on")
     await hass.async_block_till_done()
-    state = hass.states.get("light.bedroom")
-
+    state: State = hass.states.get("light.bedroom")
     assert state.last_updated == events[0].time_fired
+    assert len(state.context.id) == 26
+    # ULIDs store time to 3 decimal places compared to python timestamps
+    assert _ulid_timestamp(state.context.id) == int(
+        state.last_updated.timestamp() * 1000
+    )
+
+
+async def test_state_firing_event_matches_context_id_ulid_time(hass):
+    """Test timed_fired and the ulid have the same time."""
+    events = []
+
+    @ha.callback
+    def _event_listener(event):
+        events.append(event)
+
+    hass.bus.async_listen(EVENT_HOMEASSISTANT_STARTED, _event_listener)
```

Comment:
We can use `async_capture_events`.

https://github.com/home-assistant/core/blob/b6c7422607511f4896d7d2e28cca689413903da1/tests/common.py#L1183

Reply:
https://github.com/home-assistant/core/pull/71970

Cleaned up a few of them that were the exact same pattern